### ISSUE TITLE
[Question] How does the --fail-fast option affect the runn run behavior?

### ISSUE THREAD

**Issue by h6ah4i** [k1LoW/runn#1032]
I expected the book `3.yaml` not to run when the `--fail-fast` option is specified in the following setup. However, the command output shows that the book `3.yaml` is still being evaluated. Am I misunderstanding how this command-line option works?

```bash
❯ runn run --fail-fast './*.yaml'
.F.

1) 2.yaml 5a1a25bd510cabac2c55504c9f80759ca4ca935b
  Failure/Error: test failed on "[No Description]".steps.forceFail: condition is not true

  Condition:
    false

    │

Error: step not found: 0
```

---

### 1.yaml
```yaml
steps:
  forcePass:
    test: |
      true
```

### 2.yaml
```yaml
steps:
  forceFail:
    test: |
      false
```

### 3.yaml
```yaml
steps:
  forcePass:
    test: |
      true
```

**Comment by k2tzumi** [comment]
It might be easier to understand if you add the verbose option.

```
% runn run --fail-fast './*.yaml' --verbose
=== [No Description] (1.yaml)
    --- (forcePass) ... ok
=== [No Description] (2.yaml)
    --- (forceFail) ... fail
        Failure/Error: test failed on "[No Description]".steps.forceFail: condition is not true
        
        Condition:
          false
          
          │
          
=== [No Description] (3.yaml)
    --- (forcePass) ... ok
=== Generated by `runn new` (deptrac.yaml)


1) 2.yaml 5a1a25bd510cabac2c55504c9f80759ca4ca935b
  Failure/Error: test failed on "[No Description]".steps.forceFail: condition is not true
  
  Condition:
    false
    
    │
    
Error: step not found: 0
```

Parallel execution in goroutines when multiple runbooks are specified and already started processes will continue to run until completion; when the fail-fast option is enabled, the context is canceled to prevent new processes from starting when an error occurs, but the already started processing is not canceled.

https://github.com/k1LoW/runn/blob/main/operator.go#L1690-L1712

@k1LoW 
In order to cancel a process that has already started, I wonder if it is feasible to check for context cancellation within each goroutine and abort the process appropriately?

**Comment by k1LoW** [comment]
Hmmm. I think that since sometime ago, --fail-fast is not behaving as intended!

**Comment by k1LoW** [comment]
> In order to cancel a process that has already started, I wonder if it is feasible to check for context cancellation within each goroutine and abort the process appropriately?

Maybe we should consider it to make `--fail-fast` the intended behavior.

**Comment by k1LoW** [comment]
@h6ah4i Thank you for your report!

Fixed at v0.119.2.

```console
~/src/github.com/k1LoW/runn/tmp (main)> ls
1.yml   2.yml   3.yml
~/src/github.com/k1LoW/runn/tmp (main)> runn -v
runn version 0.119.2
~/src/github.com/k1LoW/runn/tmp (main)> runn run *.yml --fail-fast
.FS

1) tmp/2.yml f5274803c3f42fcbba30536e0cc1143e3e43811d
  Failure/Error: test failed on "[No Description]".steps.forceFail: condition is not true

  Condition:
    false

    │

Error: step not found: 0
~/src/github.com/k1LoW/runn/tmp (main)>
```
